### PR TITLE
[IMP]mail: Add custom model, id and other related info to email header

### PR DESCRIPTION
**Pull request by jas-odoo**
Task: https://www.odoo.com/web?#id=19585&view_type=form&model=project.task&menu_id=3942&action=327

Pad: https://pad.odoo.com/p/r.hvapjVGiVqrOeZU2

### CODE REVIEW

File: addons/project/project.py [hunk 1]
```diff
@@ -838,6 +838,23 @@ def message_get_suggested_recipients(self, cr, uid, ids, context=None):
                 data._message_add_suggested_recipient(recipients, partner=data.partner_id, reason=reason)
         return recipients
 
+    @api.multi
+    def message_get_email_values(self, notif_mail=None):
+        res = super(task, self).message_get_email_values(notif_mail=notif_mail)
```
Comment (atp-odoo) [inline addons/project/project.py]: ensure_one is missing ...

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -696,7 +696,10 @@ def message_get_email_values(self, notif_mail=None):
         """ Get specific notification email values to store on the notification
         mail_mail. Void method, inherit it to add custom values. """
         self.ensure_one()
-        res = dict()
+        database_uuid = self.env['ir.config_parameter'].get_param('database.uuid')
+        model = "%s-%s" % (self._model, self.id)
+        header = {'X-Odoo-Objects': model, 'X-Odoo-db-uuid': database_uuid}
+        res = {'headers': header}
```
Comment (pna-odoo) [inline addons/mail/models/mail_thread.py]: why are you assigned value here, you can direct return dict.

Comment (jas-odoo) [inline addons/mail/models/mail_thread.py]: ok

File: addons/project/project.py [hunk 3]
```diff
@@ -825,6 +825,19 @@ def message_get_suggested_recipients(self, cr, uid, ids, context=None):
...
         return recipients
 
+    @api.multi
+    def message_get_email_values(self, notif_mail=None):
+        self.ensure_one()
+        res = super(task, self).message_get_email_values(notif_mail=notif_mail)
+        headers = res.get('headers', {})
+        if headers.get('X-Odoo-Objects') and self.project_id:
+            project_info = 'project.project-%s, ' % (self.project_id.id)
+            headers['X-Odoo-Objects'] = project_info + headers['X-Odoo-Objects']
```
Comment (tde-banana-odoo) [inline addons/project/project.py]: This is not coherent with the X-Odoo-Objects meaning. Sometimes the record, sometimes its "parent". Should be stored in separate parameters.

Comment (hmo-odoo) [inline addons/project/project.py]: @tde-banana-odoo ,
here pattern of object like <model>-<record-id> eg. (project.project-2) is fixed in X-Odoo-Objects list with comma separator. so it is easily identify which record for which model. so I think, no need to other parameters.

**Pull request merged**